Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/EditorBaseComponentMode.h

```diff
@@ -41,6 +44,21 @@ namespace AzToolsFramework
             /// @attention More specific actions come later in the ordering when they are added.
             AZStd::vector<ActionOverride> PopulateActions() final;
 
+            //! Register additional actions for this component mode.
+            //! This is provided as guidance but it will not be called automatically;
+            //! you will need a system component to call it in the OnActionRegistrationHook handler.
+            static void RegisterActions() {}
+
+            //! Bind actions to appear in this component mode.
+            //! This is provided as guidance but it will not be called automatically;
+            //! you will need a system component to call it in the OnActionContextModeBindingHook handler.
+            static void BindActionsToModes() {}
+
+            //! Bind actions to appear in menus.
+            //! This is provided as guidance but it will not be called automatically;
+            //! you will need a system component to call it in the OnMenuBindingHook handler.
+            static void BindActionsToMenus() {}
```

Comment:
suggestion: This is the type of thing we should definitely make a note to document in the API for Component Mode docs when they get ported to O3DE